test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNetCore3x/BulkOperation/EFTests/BulkOperationPatchHandlersEF.cs

```diff
@@ -60,283 +60,283 @@ public APIHandlerFactoryEF(IEdmModel model, EmployeeDBContext dbContext) : base(
+                    errorMessage = ex.Message;
+                }
 
-                return ODataAPIResponseStatus.Success;
+                return status;
             }
-            catch (Exception ex)
+
+            public override IODataAPIHandler GetNestedHandler(Employee parent, string navigationPropertyName)
             {
-                errorMessage = ex.Message;
+                switch (navigationPropertyName)
+                {
+                    case "Friends":
+                        return new FriendEFPatchHandler(parent);
+                    case "NewFriends":
+                        return new NewFriendEFPatchHandler(parent);
+                    default:
+                        return null;
+                }
 
-                return ODataAPIResponseStatus.Failure;
             }
+
         }
 
-        public override ODataAPIResponseStatus TryDelete(IDictionary<string, object> keyValues, out string errorMessage)
+        internal class FriendEFPatchHandler : ODataAPIHandler<Friend>
         {
-            errorMessage = string.Empty;
+            Employee employee;
+            public FriendEFPatchHandler(Employee employee)
+            {
+                this.employee = employee;
+            }
 
-            try
+            public override ODataAPIResponseStatus TryCreate(IDictionary<string, object> keyValues, out Friend createdObject, out string errorMessage)
             {
-                var id = keyValues.First().Value.ToString();
-                var friend = employee.Friends.First(x => x.Id == Int32.Parse(id));
+                createdObject = null;
+                errorMessage = string.Empty;
 
-                employee.Friends.Remove(friend);
+                try
+                {
+                    createdObject = new Friend();
+                    employee.Friends.Add(createdObject);
 
-                return ODataAPIResponseStatus.Success;
-            }
-            catch (Exception ex)
-            {
-                errorMessage = ex.Message;
+                    return ODataAPIResponseStatus.Success;
+                }
+                catch (Exception ex)
+                {
+                    errorMessage = ex.Message;
 
-                return ODataAPIResponseStatus.Failure;
+                    return ODataAPIResponseStatus.Failure;
+                }
             }
-        }
 
-        public override ODataAPIResponseStatus TryGet(IDictionary<string, object> keyValues, out Friend originalObject, out string errorMessage)
-        {
-            ODataAPIResponseStatus status = ODataAPIResponseStatus.Success;
-            errorMessage = string.Empty;
-            originalObject = null;
-
-            try
+            public override ODataAPIResponseStatus TryDelete(IDictionary<string, object> keyValues, out string errorMessage)
             {
-                var id = keyValues["Id"].ToString();
-                if (employee.Friends == null)
+                errorMessage = string.Empty;
```

Comment:
Curious why we initialize `errorMessage` to empty instead of null yet that is the default value for a string

Reply:
I have seen the same initialization in `DeltaSet<T>`. I have no idea why it's so. My thinking is the `DataModificationExceptionType` can have an empty message.
cc @ElizabethOkerio

Reply:
From docs, both `null` and `Empty` are recommended https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/strings/#declaring-and-initializing-strings